supfile.go

```diff
@@ -13,21 +13,66 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+// EnvVar represents an environment variable
+type EnvVar struct {
+	Key   string
+	Value string
+}
+
+func (e EnvVar) String() string {
+	return e.Key + `=` + e.Value
+}
+
+// AsExport returns the environment variable as a bash export statement
+func (e EnvVar) AsExport() string {
+	return `export ` + e.Key + `="` + e.Value + `";`
+}
+
+// EnvList is a list of environment variables that maps to a YAML map,
+// but maintains order, enabling late variables to reference early variables.
+type EnvList []EnvVar
+
+func (e *EnvList) UnmarshalYAML(ufn func(interface{}) error) error {
+	d := []yaml.MapItem{}
+
+	err := ufn(&d)
+	if err != nil {
+		return err
+	}
+
+	*e = make(EnvList, 0, len(d))
```

Comment:
Hm, don't think that would work. It would create a list of `len(d)` with zero values for `EnvVar` (ie blank `.Key`/`.Value`). 

http://play.golang.org/p/npEaCZxmdE